### PR TITLE
bump containerd/cgroups 5fbad35c2a7e855762d3c60f2e474ffcad0d470a

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/containerd/cgroups/compare/c4b9ac5c7601384c965b9646fc515884e091ebb9...5fbad35c2a7e855762d3c60f2e474ffcad0d470a

- containerd/cgroups#82 Add go module support
- containerd/cgroups#96 Move metrics proto package to stats/v1
- containerd/cgroups#97 Allow overriding the default /proc folder in blkioController
- containerd/cgroups#98 Allows ignoring memory modules
- containerd/cgroups#99 Add Go 1.13 to Travis
- containerd/cgroups#100 stats/v1: export per-cgroup stats

### CODE REVIEW

**Comment by thaJeztah** [comment]
Ah, fun; so this requires a more recent version of libcontainerd; 

```
libcontainerd/types/types_linux.go:16:11: undefined: cgroups.Metrics
```

**Comment by thaJeztah** [comment]
Failures on Windows RS1 is interesting; looks like we're trying to use a unix-socket on windows;
https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-40154/runs/5/nodes/52/log/?start=0

```
[2019-10-30T13:19:27.378Z] --- FAIL: TestInfoDebug (0.02s)
[2019-10-30T13:19:27.378Z]     info_test.go:75: [d757c903a4332] failed to start daemon with arguments [--containerd /var/run/docker/containerd/containerd.sock --data-root D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoDebug\d757c903a4332\root --exec-root C:\windows\TEMP\dxr\d757c903a4332 --pidfile D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoDebug\d757c903a4332\docker.pid --userland-proxy=true --containerd-namespace d757c903a4332 --containerd-plugins-namespace d757c903a4332p --host unix://C:\windows\TEMP\docker-integration\d757c903a4332.sock --storage-driver overlay2 --debug] : protocol not available
[2019-10-30T13:19:27.378Z] === RUN   TestInfoInsecureRegistries
[2019-10-30T13:19:27.378Z] --- FAIL: TestInfoInsecureRegistries (0.00s)
[2019-10-30T13:19:27.378Z]     info_test.go:100: [d3dab042d8737] failed to start daemon with arguments [--containerd /var/run/docker/containerd/containerd.sock --data-root D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoInsecureRegistries\d3dab042d8737\root --exec-root C:\windows\TEMP\dxr\d3dab042d8737 --pidfile D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoInsecureRegistries\d3dab042d8737\docker.pid --userland-proxy=true --containerd-namespace d3dab042d8737 --containerd-plugins-namespace d3dab042d8737p --host unix://C:\windows\TEMP\docker-integration\d3dab042d8737.sock --debug --storage-driver overlay2 --insecure-registry=192.168.1.0/24 --insecure-registry=insecurehost.com:5000] : protocol not available
[2019-10-30T13:19:27.378Z] === RUN   TestInfoRegistryMirrors
[2019-10-30T13:19:27.378Z] --- FAIL: TestInfoRegistryMirrors (0.01s)
[2019-10-30T13:19:27.378Z]     info_test.go:124: [de65b11b7ae5d] failed to start daemon with arguments [--containerd /var/run/docker/containerd/containerd.sock --data-root D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoRegistryMirrors\de65b11b7ae5d\root --exec-root C:\windows\TEMP\dxr\de65b11b7ae5d --pidfile D:\gopath\src\github.com\docker\docker\bundles\tmp\TestInfoRegistryMirrors\de65b11b7ae5d\docker.pid --userland-proxy=true --containerd-namespace de65b11b7ae5d --containerd-plugins-namespace de65b11b7ae5dp --host unix://C:\windows\TEMP\docker-integration\de65b11b7ae5d.sock --debug --storage-driver overlay2 --registry-mirror=https://192.168.1.2 --registry-mirror=http://registry.mirror.com:5000] : protocol not available
```


```
--host unix://C:\windows\TEMP\docker-integration\d757c903a4332.sock
--storage-driver overlay2 --debug
protocol not available
```


Other failure is:

```
[2019-10-30T14:19:44.318Z]     --- FAIL: TestDockerSuite/TestBuildDockerignoreExceptions (59.01s)
[2019-10-30T14:19:44.318Z]         docker_cli_build_test.go:2351: assertion failed:
[2019-10-30T14:19:44.318Z]             Command:  d:\CI\PR-40154\5\binary\docker.exe build -t testbuilddockerignoreexceptions .
[2019-10-30T14:19:44.318Z]             ExitCode: 1
[2019-10-30T14:19:44.318Z]             Error:    exit status 1
[2019-10-30T14:19:44.318Z]             Stdout:   Sending build context to Docker daemon  8.704kB
[2019-10-30T14:19:44.318Z]
[2019-10-30T14:19:44.318Z]             Step 1/14 : FROM busybox
[2019-10-30T14:19:44.318Z]              ---> 0261edd4d45e
[2019-10-30T14:19:44.318Z]             Step 2/14 : ADD . /bla
[2019-10-30T14:19:44.318Z]              ---> e0f30eafb58e
[2019-10-30T14:19:44.318Z]             Step 3/14 : RUN sh -c "[[ -f /bla/src/x.go ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 3a6818356d2e
[2019-10-30T14:19:44.318Z]             Removing intermediate container 3a6818356d2e
[2019-10-30T14:19:44.318Z]              ---> 1f27a9e725ec
[2019-10-30T14:19:44.318Z]             Step 4/14 : RUN sh -c "[[ -f /bla/Makefile ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in becac854582c
[2019-10-30T14:19:44.318Z]             Removing intermediate container becac854582c
[2019-10-30T14:19:44.318Z]              ---> 7de50716cf78
[2019-10-30T14:19:44.318Z]             Step 5/14 : RUN sh -c "[[ ! -e /bla/src/_vendor ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in e093121ac05d
[2019-10-30T14:19:44.318Z]             Removing intermediate container e093121ac05d
[2019-10-30T14:19:44.318Z]              ---> 4c096849cc23
[2019-10-30T14:19:44.318Z]             Step 6/14 : RUN sh -c "[[ ! -e /bla/.gitignore ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 2b75521a3e8c
[2019-10-30T14:19:44.318Z]             Removing intermediate container 2b75521a3e8c
[2019-10-30T14:19:44.318Z]              ---> f3d5aaf6c5ff
[2019-10-30T14:19:44.318Z]             Step 7/14 : RUN sh -c "[[ ! -e /bla/README.md ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 69716f78fc7f
[2019-10-30T14:19:44.318Z]             Removing intermediate container 69716f78fc7f
[2019-10-30T14:19:44.318Z]              ---> 78ba8722b7d5
[2019-10-30T14:19:44.318Z]             Step 8/14 : RUN sh -c "[[  -e /bla/dir/dir/foo ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 2461d8c9be92
[2019-10-30T14:19:44.318Z]             Removing intermediate container 2461d8c9be92
[2019-10-30T14:19:44.318Z]              ---> 770346c0fc32
[2019-10-30T14:19:44.318Z]             Step 9/14 : RUN sh -c "[[ ! -e /bla/dir/foo1 ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 223a625bd942
[2019-10-30T14:19:44.318Z]             Removing intermediate container 223a625bd942
[2019-10-30T14:19:44.318Z]              ---> 5b73811b072a
[2019-10-30T14:19:44.318Z]             Step 10/14 : RUN sh -c "[[ -f /bla/dir/e ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in 6d26bd279f1a
[2019-10-30T14:19:44.318Z]             Removing intermediate container 6d26bd279f1a
[2019-10-30T14:19:44.318Z]              ---> 3d4e6b8b197e
[2019-10-30T14:19:44.318Z]             Step 11/14 : RUN sh -c "[[ -f /bla/dir/e-dir/foo ]]"
[2019-10-30T14:19:44.318Z]              ---> Running in c6ca7aa4cbc4
[2019-10-30T14:19:44.318Z]
[2019-10-30T14:19:44.319Z]             Stderr:   re-exec error: exit status 1: output: time="2019-10-30T13:34:53Z" level=error msg="hcsshim::ImportLayer - failed failed in Win32: There is not enough space on the disk. (0x70)" error="hcsshim::ImportLayer - failed failed in Win32: There is not enough space on the disk. (0x70)" importFolderPath="D:\\CI\\PR-40154\\5\\daemon\\tmp\\hcs262570275" path="\\\\?\\D:\\CI\\PR-40154\\5\\daemon\\windowsfilter\\552078b33187a91163be0570d75c4780a6c0ebb46442fc4118a86b1d9fca90cd"
[2019-10-30T14:19:44.319Z]             hcsshim::ImportLayer - failed failed in Win32: There is not enough space on the disk. (0x70)
[2019-10-30T14:19:44.319Z]
[2019-10-30T14:19:44.319Z]
[2019-10-30T14:19:44.319Z]             Failures:
[2019-10-30T14:19:44.319Z]             ExitCode was 1 expected 0
[2019-10-30T14:19:44.319Z]             Expected no error
```

Could be the host having issues;

```
hcsshim::ImportLayer - failed failed in Win32: There is not enough space on the disk. (0x70)
```

**Comment by thaJeztah** [comment]
Opened https://github.com/moby/moby/pull/40155 to skip for now, but we need to look into this, and especially Windows RS5 not reporting failures

@ddebroy @vikramhh @andrewhsu

**Comment by thaJeztah** [comment]
ping @dmcgowan @jterry75 @kolyshkin PTAL

**Comment by vikramhh** [comment]
@thaJeztah - Windows RS5 not reporting failures has been root caused and I have a fix in mind. Can we take in https://github.com/moby/moby/pull/39998 before that though - will avoid some duplication of work as the fix will need to be reworked again for https://github.com/moby/moby/pull/39998.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by vikramhh** [comment]
LGTM

**Review by tiborvass — APPROVED** [review]

File: daemon/daemon_unix.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 	"strings"
 	"time"
 
-	containerd_cgroups "github.com/containerd/cgroups"
+	v1 "github.com/containerd/cgroups/stats/v1"
```
Comment (kolyshkin) [inline daemon/daemon_unix.go]: this is kinda weird to call this `v1`. maybe just `cgroups`?

File: libcontainerd/types/types_linux.go [hunk 2]
```diff
@@ -3,7 +3,7 @@ package types // import "github.com/docker/docker/libcontainerd/types"
 import (
 	"time"
 
-	"github.com/containerd/cgroups"
+	v1 "github.com/containerd/cgroups/stats/v1"
```
Comment (kolyshkin) [inline libcontainerd/types/types_linux.go]: ditto

File: daemon/daemon_unix.go (hunk 1 above)
Comment (AkihiroSuda) [reply daemon/daemon_unix.go]: "statsv1" (v2 will be added soon)

Comment (thaJeztah) [reply daemon/daemon_unix.go]: Opened https://github.com/moby/moby/pull/40167 to rename to `statsV1`

**Pull request merged**